magda-sleuther-framework/src/setupWebhookEndpoint.ts

```diff
@@ -13,20 +13,21 @@ export default function setupWebhookEndpoint(
     registry: Registry
 ) {
     const server = options.express();
-    server.use(require("body-parser").json({limit: '50mb'}));
+    server.use(require("body-parser").json({ limit: "50mb" }));
 
     server.post(
         "/hook",
         (request: express.Request, response: express.Response) => {
             const payload = request.body;
 
-            const promises: Promise<
-                void
-            >[] = payload.records.map((record: Record) =>
-                options.onRecordFound(record, registry)
+            const promises: (() => Promise<void>)[] = payload.records.map(
+                (record: Record) => () =>
+                    options.onRecordFound(record, registry)
+            );
+            const megaPromise = promises.reduce(
```

Comment:
Consider putting the records in an `AsyncPage` (e.g. `AsyncPage.single(payload.records)`) and then use `forEachAsync` to allow a configurable number of simultaneous records here too.